### PR TITLE
Some properties of upTo and downFrom

### PR DESCRIPTION
**Pull request by Taneb**
It's nice to show that the two lists have the same actual contents, just in a different order

### CODE REVIEW

**Review by JacquesCarette — APPROVED** [review]

File: src/Data/List/Properties.agda [hunk 1]
```diff
@@ -1173,6 +1187,31 @@ reverse-foldl : ∀ (f : B → A → B) b xs →
                 foldl f b (reverse xs) ≡ foldr (flip f) b xs
 reverse-foldl f b xs = foldl-ʳ++ f b xs
 
+------------------------------------------------------------------------
+-- reverse, applyUpTo, and applyDownFrom
+
+applyUpTo-applyDownFrom : ∀ (f : ℕ → A) n → reverse (applyUpTo f n) ≡ applyDownFrom f n
```
Comment (MatthewDaggitt) [inline src/Data/List/Properties.agda:1193]: Name should be `reverse-applyUpTo`

File: src/Data/List/Properties.agda [hunk 2]
```diff
@@ -1173,6 +1187,31 @@ reverse-foldl : ∀ (f : B → A → B) b xs →
...
+applyUpTo-applyDownFrom f zero = refl
+applyUpTo-applyDownFrom f (suc n) = begin
+  reverse (f 0 ∷ applyUpTo (f ∘ suc) n)  ≡⟨ reverse-++ [ f 0 ] (applyUpTo (f ∘ suc) n) ⟩
+  reverse (applyUpTo (f ∘ suc) n) ∷ʳ f 0 ≡⟨ cong (_∷ʳ f 0) (applyUpTo-applyDownFrom (f ∘ suc) n) ⟩
+  applyDownFrom (f ∘ suc) n ∷ʳ f 0       ≡⟨ applyDownFrom-∷ʳ f n ⟩
+  applyDownFrom f (suc n)                ∎
+
+upTo-downFrom : ∀ n → reverse (upTo n) ≡ downFrom n
```
Comment (MatthewDaggitt) [inline src/Data/List/Properties.agda:1201]: Name should be `reverse-upTo`

File: src/Data/List/Properties.agda [hunk 3]
```diff
@@ -1173,6 +1187,31 @@ reverse-foldl : ∀ (f : B → A → B) b xs →
...
+upTo-downFrom = applyUpTo-applyDownFrom id
+
+applyDownFrom-applyUpTo : ∀ (f : ℕ → A) n → reverse (applyDownFrom f n) ≡ applyUpTo f n
```
Comment (MatthewDaggitt) [inline src/Data/List/Properties.agda:1204]: Name should be `reverse-applyDownFrom`

File: src/Data/List/Properties.agda [hunk 4]
```diff
@@ -1173,6 +1187,31 @@ reverse-foldl : ∀ (f : B → A → B) b xs →
...
+applyDownFrom-applyUpTo f zero = refl
+applyDownFrom-applyUpTo f (suc n) = begin
+  reverse (f n ∷ applyDownFrom f n)  ≡⟨ reverse-++ [ f n ] (applyDownFrom f n) ⟩
+  reverse (applyDownFrom f n) ∷ʳ f n ≡⟨ cong (_∷ʳ f n) (applyDownFrom-applyUpTo f n) ⟩
+  applyUpTo f n ∷ʳ f n               ≡⟨ applyUpTo-∷ʳ f n ⟩
+  applyUpTo f (suc n)                ∎
+
+downFrom-upTo : ∀ n → reverse (downFrom n) ≡ upTo n
```
Comment (MatthewDaggitt) [inline src/Data/List/Properties.agda:1212]: Name should be `reverse-downFrom`

**Review by MatthewDaggitt — CHANGES_REQUESTED** [review]
Otherwise looks good

File: CHANGELOG.md [hunk 5]
```diff
@@ -140,6 +140,18 @@ Additions to existing modules
...
   ```
 
+* In `Data.List.Properties`:
+  ```agda
+  applyUpTo-∷ʳ          : applyUpTo f n ∷ʳ f n ≡ applyUpTo f (suc n)
+  applyDownFrom-∷ʳ      : applyDownFrom (f ∘ suc) n ∷ʳ f 0 ≡ applyDownFrom f (suc n)
+  upTo-∷ʳ               : upTo n ∷ʳ n ≡ upTo (suc n)
+  downFrom-∷ʳ           : applyDownFrom suc n ∷ʳ 0 ≡ downFrom (suc n)
+  reverse-applyUpTo     : reverse (applyUpTo f n) ≡ applyDownFrom f n
+  reverse-downFrom      : reverse (upTo n) ≡ downFrom n
```
Comment (MatthewDaggitt) [inline CHANGELOG.md:150]: Typo

**Review by MatthewDaggitt — APPROVED** [review]

**Pull request merged**